ddtrace/propagation/http.py

```diff
@@ -543,6 +543,16 @@ class _TraceContext:
         ``_dd.p.`` prefix = ``t.``
     """
 
+    # incoming traceparent header name and tracestate header name may be in any case (upper, lower, mixed)
+    # they may also have leading or trailing whitespace
+    @staticmethod
+    def _extract_header_value_tracecontext(possible_header_names, headers, default=None):
+        # type: (FrozenSet[str], Dict[str, str], Optional[str]) -> Optional[str]
+        headers_lower = {k.lower().strip(): v for k, v in headers.items()}
```

Comment:
don't these get normalized already below in `HTTPPropagator.extract`?

Reply:
if we do this again here, the performance is going to be **bad**

Reply:
In that case, should I just change the headers in HTTPPropagator.extract to be lowercase and have no white space? The issue I see with that, is that `b3 single header` is supposed to have whitespace.

Reply:
yeah, just add `.strip()` there